tests/graph/tests/integration/graph/edge-removal/helpers.ts

```diff
@@ -125,24 +132,29 @@ export async function setInitialState(context: Context, config: TestConfig, asse
         id: '1',
         attributes: { name: 'Chris' },
       },
-    });
-    john = store.createRecord('user', { name: 'John', bestFriends: isMany ? [chris] : chris });
+    } as SingleResourceDocument);
+    john = store.createRecord('user', { name: 'John', bestFriends: isMany ? [chris] : chris }) as UserRecord;
     johnIdentifier = recordIdentifierFor(john);
   }
 
   if (config.dirtyLocal) {
     if (isMany) {
-      let friends = await john.bestFriends;
+      let friends: UserRecord[] = await (john.bestFriends as unknown as Promise<UserRecord[]>);
       friends.push(chris);
-      friends = await chris.bestFriends;
-      friends.push(john);
+      if (config.inverseNull) {
+        friends = await (chris.bestFriends as unknown as Promise<UserRecord[]>);
+        friends.push(john);
+      }
     } else {
+      // @ts-expect-error
       john.bestFriends = chris;
+      // @ts-expect-error
       chris.bestFriends = john;
     }
   }
 
-  await settled();
+  // give ourselves a tick in case there was async work
+  await Promise.resolve();
```

Comment:
This change came from copy-pasta. Should I keep it?

Reply:
I think this should be fine